### ISSUE TITLE
App crashes sometimes when accessing ImapClient MimeMessage HtmlBody

### ISSUE THREAD

**Issue by barruka** [jstedfast/MimeKit#890]
The app crashes sometimes when iterating messages of imap server and accessing the HtmlBody property of the MimeMessage.

I cannot provide the HTML code that is causing the crash, because it is corporation confidential, but I have tested the app just changing the PackageReference version and I found that with these versions the app crases just when accessing .HtmlBody:
MailKit & MimeKit versions: 3.5.0, 3.4.3, 3.4.2, 3.4.1, 3.4.0

Otherwise, just changing the PackageReference version to these ones, withou any other change the app works fine:
MailKit & MimeKit versions: 3.3.0, 3.2.0, 3.1.0, 3.0.0

**Platform (please complete the following information):**
 - OS: linux docker container running .net 7.0-alpine
 - .NET Runtime: 7.0.3
 - MimeKit Version: 3.5.0, 3.4.3, 3.4.2, 3.4.1, 3.4.0

**Comment by jstedfast** [comment]
Can you get me a StackTrace at least?

Also it would really be helpful to have the message (even if you anonymize the content). MimeKit doesn't do anything with the HTML per se (other than maybe converting the raw bytes into a .NET `string` via something like `Encoding.UTF8.GetString(bytes)`.

Let me give you an example of how you could anonymize:

```
From: Bob <bob@acme-anvils.com>
To: Alice <alice@acme-anvils.com>
Subject: Have you seen the latest anvil?
Message-Id: <01.1023759.3027@acme-anvils.com>
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="=_Next_Alternative_001"

--=_Next_Alternative_001
Content-Type: text/plain; charset="us-ascii"

Hey Alice!

Have you seen the newest anvils that were just brought in this morning?

Wow!

These are sure to get that wiley coyote!

-- Bob
--=_Next_Alternative_001
Content-Type: text/html; charset="us-ascii"

<html><body><p>Hey Alice!</p><p>Have you seen the newest anvils that were just brought in this morning?</p>
<p>Wow!</p><p>These are sure to get that wiley coyote!</p><p>-- Bob</p></body></html>
--=_Next_Alternative_001--
```

This could be anonymized like this:

```
From: xxx <xxx@xxxxxxxxxx.com>
To: xxxxx <xxxxx@xxxxxxxxxx.com>
Subject: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
Message-Id: <01.1023759.3027@xxxxxxxxxx.com>
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="=_Next_Alternative_001"

--=_Next_Alternative_001
Content-Type: text/plain; charset="us-ascii"

Confidential message content was here.
--=_Next_Alternative_001
Content-Type: text/html; charset="us-ascii"

Confidential HTML message content was here.
--=_Next_Alternative_001--
```

**Comment by barruka** [comment]
I don't know how can I catch the StackTrace because the app just crashes and the crash doen't raise the try catch so the log call it's not executing.

I can see the csw "... DownloadEmails_IMAP Email downloaded..." with other emails but not with the email that is causing the app to crash.

But just downgrading the PackageReference version of both MimeKit and MailKit from 3.5.0 to 3.3.0  it works fine


                foreach (var uid in uids)
                {
                    var message = client.Inbox.GetMessage(uid);
                    try
                    {

                             // ...

                            emails.Add(new Email(new EmailCreateDto(
                                                                    Id: Guid.NewGuid().ToString(),
                                                                    ImapUid: uid.ToString().Left(255),
                                                                    CreatedAt: dateFromHeader == null ? date : dateFromHeader.Value,
                                                                    FromEmailAddress: "" + message.From.Mailboxes.FirstOrDefault()?.Address?.ToLower(),
                                                                    FromEmailName: message.From.Mailboxes.FirstOrDefault()?.Name,
                                                                    ToEmailAddress: "" + toAddress,
                                                                    ToEmailName: message.To.Mailboxes.FirstOrDefault()?.Name,
                                                                    DeliveredToEmailAddress: message.Headers["Delivered-To"]?.ToString()?.ToLower(),
                                                                    Subject: "" + message.Subject,
                                                                    BodyText: message.TextBody,
                                                                    BodyHtml: message.HtmlBody,
                                                                    Attachments: attachments
                                                                   )
                                                )
                                      );
                       

                        Console.WriteLine($"{message.Date.LocalDateTime} DownloadEmails_IMAP Email downloaded: {message.Subject}");
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine($"{message.Date.LocalDateTime} DownloadEmails_IMAP ERROR descargando email {ex.Message}");
                        _logger.LogError(ex, JsonSerializer.Serialize(message));
                    }
                }
`

**Comment by jstedfast** [comment]
I'm unable to reproduce the crash :(

**Comment by barruka** [comment]
I'm very sorry for not beeing helpfull .

**Comment by jstedfast** [comment]
The thing to do to help narrow down the problem is to add more logging.

* Does it crash in `client.Inbox.GetMessage(uid);`?
* Does it crash in the `// ...` section of code?
* Instead of using expressions for each of the EmailCreateDto .ctor arguments, try using temp variables to get the values *before* the `new EmailCreateDto (...` call and putting logging between each of those statements to help determine if one of those lines is crashing.

**Comment by jstedfast** [comment]
Another thing to keep in mind is that Console.WriteLine() will not necessarily get to flush its output on a crash. Instead, use Console.Error.WriteLine() because that will flush the output on each call.

**Comment by barruka** [comment]
Yeah, I break down the code and I used temp variables before creating the DTO to see which field was causing the problem. I removed the DTO creation and I checked that when the code has "var html = message.HtmlBody;" it crashes. I commented the line and the the app works and not crashes.

I did some random tests and I figured out that the container running the app in production was MEMORY LIMITED to 96 MB. Next days I'll try to get at least 256 MB assigned to the container and try to run the app bumping it up to 3.5.0. 

I'll keep you in the loop.

(For now, with 3.3.0 it's working fine.)
